assets/js/modules/tagmanager/datastore/containers.js

```diff
@@ -0,0 +1,171 @@
+	},
+	controlCallback: ( { accountID } ) => {
+		// Always request both contexts to prevent filtering on server.
+		// TODO: Remove `usageContext` param when legacy component is removed and datapoint
+		// defaults to returning all containers if no context is provided.
+		const usageContext = [ CONTEXT_WEB, CONTEXT_AMP ];
+		return API.get( 'modules', 'tagmanager', 'containers', { accountID, usageContext }, { useCache: false } );
+	},
+	reducerCallback: ( state, containers, { accountID } ) => {
+		return {
+			...state,
+			containers: {
+				...state.containers,
+				[ accountID ]: [ ...containers ],
+			},
+		};
+	},
+} );
+
+const fetchCreateContainerStore = createFetchStore( {
+	baseName: 'createContainer',
+	argsToParams( accountID, usageContext ) {
+		invariant( isValidAccountID( accountID ), 'A valid accountID is required to create a container.' );
+		invariant( isValidUsageContext( usageContext ), 'A valid usageContext is required to create a container.' );
+
+		return { accountID, usageContext };
+	},
+	controlCallback: ( { accountID, usageContext } ) => {
+		return API.set( 'modules', 'tagmanager', 'create-container', { accountID, usageContext } );
+	},
+	reducerCallback( state, container, { accountID } ) {
+		return {
+			...state,
+			containers: {
+				...state.containers,
+				[ accountID ]: [
+					...( state.containers[ accountID ] || [] ),
+					container,
+				],
+			},
+		};
+	},
+} );
+
+const BASE_INITIAL_STATE = {
+	containers: {},
+};
+
+const baseActions = {
+	/**
+	 * Creates a new Tag Manager container in the given account.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {string} accountID    Google Tag Manager account ID.
+	 * @param {string} usageContext Container usage context. (Either 'web', or 'amp')
+	 * @return {Object} Object with `response` and `error`.
+	 */
+	*createContainer( accountID, usageContext ) {
+		invariant( isValidAccountID( accountID ), 'A valid accountID is required to create a container.' );
+		invariant( isValidUsageContext( usageContext ), 'A valid usageContext is required to create a container.' );
+
+		const { response, error } = yield fetchCreateContainerStore.actions.fetchCreateContainer( accountID, usageContext );
+
+		return { response, error };
+	},
+};
+
+const baseResolvers = {
+	*getContainers( accountID ) {
+		const { select } = yield Data.commonActions.getRegistry();
+
+		if ( ! select( STORE_NAME ).getContainers( accountID ) ) {
+			yield fetchGetContainersStore.actions.fetchGetContainers( accountID );
+		}
+	},
+};
+
+const baseSelectors = {
+	/**
+	 * Gets the containers for a given account.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} state          Data store's state.
+	 * @param {string} accountID      Account ID to get containers for.
+	 * @param {string} [usageContext] Usage context of containers to filter by.
+	 * @return {(Array|undefined)} Array of containers, or `undefined` if not loaded yet.
+	 */
+	getContainers( state, accountID, usageContext ) {
```

Comment:
See my other comment, we may want to put `usageContext` into an object here. It's an optional filter argument, so I think that would make sense. Or we could remove this parameter from here entirely, may make more sense, and then introduce `getWebContainers` and `getAMPContainers` selectors which would rely on `getContainers` and then filter accordingly.

Reply:
I like the idea of the dedicated context-specific selectors 👍